java/src/main/java/com/logicalclocks/hsfs/engine/HudiEngine.java

```diff
@@ -208,15 +209,20 @@ private FeatureGroupCommit getLastCommitMetadata(SparkSession sparkSession, Stri
 
   private Map<String, String> setupHudiReadOpts(Long startTimestamp, Long endTimestamp,
                                                 Map<String, String> readOptions) {
-    Map<String, String> hudiArgs = new HashMap<String, String>();
+    Map<String, String> hudiArgs = new HashMap<>();
 
     if (startTimestamp != null) {
       hudiArgs.put(HUDI_BEGIN_INSTANTTIME_OPT_KEY, timeStampToHudiFormat(startTimestamp));
     } else {
       hudiArgs.put(HUDI_BEGIN_INSTANTTIME_OPT_KEY, timeStampToHudiFormat(0L));
     }
 
-    hudiArgs.put(HUDI_END_INSTANTTIME_OPT_KEY, timeStampToHudiFormat(endTimestamp));
+    if (endTimestamp != null) {
+      hudiArgs.put(HUDI_END_INSTANTTIME_OPT_KEY, timeStampToHudiFormat(endTimestamp));
+    } else {
+      hudiArgs.put(HUDI_END_INSTANTTIME_OPT_KEY, timeStampToHudiFormat(System.currentTimeMillis()));
```

Comment:
Why did this change?

Reply:
When returning the training dataset query, the end timestamp is empty. But Hudi needs an "end" timestamp to decide which commits to include. I think that for batch inference we should include all the commits.